schema_poc/producer.py

```diff
@@ -0,0 +1,43 @@
+from pika import BlockingConnection, ConnectionParameters, BasicProperties
+
+import json
+from io import StringIO
+from schema_registry import RESPONSE_KEY_SCHEMA, RESPONSE_KEY_VERSION, SchemaRegistry
+
+from fastavro import json_writer, parse_schema
+
+
+class Producer:
+    def __init__(self, schema_registry: SchemaRegistry):
+        self._schema_registry = schema_registry
+
+    def prepare_message(self, message):
+        subject = "plate-map-sample"
+        write_schema_response = self._schema_registry.get_latest_schema(subject)
+        write_schema_version = write_schema_response[RESPONSE_KEY_VERSION]
+        write_schema_obj = json.loads(write_schema_response[RESPONSE_KEY_SCHEMA])
+
+        write_schema = parse_schema(write_schema_obj)
+        string_writer = StringIO()
+        json_writer(string_writer, write_schema, message)
+
+        prepared_message = {"subject": subject, "version": write_schema_version, "message": string_writer.getvalue()}
+
+        return prepared_message
+
+    def send_message(self, message_and_info, exchange, queue):
+        subject = message_and_info["subject"]
+        version = message_and_info["version"]
+        message = message_and_info["message"]
```

Comment:
Would it be worth the `prepare_message` method turning the properties into a `BasicProperties` object and returning that?  Otherwise you're mixing up the properties with the message body in the return type from that method and then having to parse them all back out to turn them into `BasicProperties` in the end.